atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/KeyValueServiceDataTrackerTest.java

```diff
@@ -0,0 +1,306 @@
+            });
+
+            executor.execute(() -> tracker.recordTableAgnosticRead(KVS_METHOD_NAME_4, BYTES_READ_1));
+            executor.execute(() -> tracker.recordTableAgnosticRead(KVS_METHOD_NAME_5, BYTES_READ_2));
+
+            executor.execute(tracker::getReadInfo);
+            executor.execute(tracker::getReadInfo);
+            executor.execute(tracker::getReadInfo);
+
+            executor.execute(tracker::getReadInfoByTable);
+            executor.execute(tracker::getReadInfoByTable);
+            executor.execute(tracker::getReadInfoByTable);
+        }
+
+        executor.shutdown();
+        assertThat(executor.awaitTermination(15, TimeUnit.SECONDS)).isTrue();
+
+        assertThat(tracker.getReadInfo())
+                .isEqualTo(createTransactionReadInfo(
+                        CONCURRENCY_ROUNDS * (5 * BYTES_READ_1 + 6 * BYTES_READ_2 + 4 * BYTES_READ_3),
+                        CONCURRENCY_ROUNDS * 12,
+                        createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_2)));
+
+        assertThat(tracker.getReadInfoByTable())
+                .containsExactlyEntriesOf(ImmutableMap.of(
+                        TABLE_1,
+                        createTransactionReadInfo(
+                                CONCURRENCY_ROUNDS * (BYTES_READ_1 + BYTES_READ_2 + 3 * BYTES_READ_3),
+                                CONCURRENCY_ROUNDS * 5,
+                                createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_2)),
+                        TABLE_2,
+                        createTransactionReadInfo(
+                                CONCURRENCY_ROUNDS * (BYTES_READ_1 + 2 * BYTES_READ_2 + BYTES_READ_3),
+                                CONCURRENCY_ROUNDS * 3,
+                                createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_3)),
+                        TABLE_3,
+                        createTransactionReadInfo(
+                                CONCURRENCY_ROUNDS * (2 * BYTES_READ_1 + 2 * BYTES_READ_2),
+                                CONCURRENCY_ROUNDS * 2,
+                                createMaximumBytesKvsCallInfo(BYTES_READ_2, KVS_METHOD_NAME_1))));
+    }
+
+    private static TransactionReadInfo createTransactionReadInfo(long bytesRead, int kvsCalls) {
+        return ImmutableTransactionReadInfo.builder()
+                .bytesRead(bytesRead)
+                .kvsCalls(kvsCalls)
+                .build();
+    }
+
+    private static TransactionReadInfo createTransactionReadInfo(
+            long bytesRead, long kvsCalls, KvsCallReadInfo maximumBytesKvsCallInfo) {
+        return ImmutableTransactionReadInfo.builder()
+                .bytesRead(bytesRead)
+                .kvsCalls(kvsCalls)
+                .maximumBytesKvsCallInfo(maximumBytesKvsCallInfo)
+                .build();
+    }
+
+    private static KvsCallReadInfo createMaximumBytesKvsCallInfo(long bytesRead, String kvsMethodName) {
+        return ImmutableKvsCallReadInfo.builder()
```

Comment:
side-note: you can add a static of method (or generate it using immutables) to immutable objects, which means you'll get this helper method for free. 

I don't think I'd do that for the TransactionReadInfo, for the sole reason that you're dealing with 2 longs and the builder names help manage that - but for KvsCallReadInfo it'd simplify some code a little.